### PR TITLE
Argument support for callable dictionary

### PR DESCRIPTION
**Pull request by Nek-**
This is a feature from the original project. Even if it's under MIT license I'd like @NiR- to confirm he is ok with the idea that we take his changes before the merge.

### CODE REVIEW

**Comment by athos7933** [comment]
I guess missing changes on documentation in advances usage => Callable dictionary ? Right ?

**Comment by Nek-** [comment]
@athos7933 you're right. The original project also doesn't mention any doc about this, I'm just back-porting a new feature.

**Comment by NiR-** [comment]
@Nek- As you said this work is under MIT license, so I really don't care, whatever you want to do with it.

File: spec/Knp/DictionaryBundle/Dictionary/CallableDictionarySpec.php [hunk 1]
```diff
@@ -119,8 +128,9 @@ public function execution()
     {
         if (false === $this->executed) {
             $this->executed = true;
+            $args = func_get_args();
 
-            return  [
+            return !empty($args) ? current($args) : [
```
Comment (Awkan) [inline spec/Knp/DictionaryBundle/Dictionary/CallableDictionarySpec.php]: `\`

File: src/Knp/DictionaryBundle/Dictionary/CallableDictionary.php [hunk 2]
```diff
@@ -120,7 +125,7 @@ protected function hydrate()
             return;
         }
 
-        $values = call_user_func($this->callable);
+        $values = call_user_func_array($this->callable, $this->callableArgs);
```
Comment (Awkan) [inline src/Knp/DictionaryBundle/Dictionary/CallableDictionary.php]: `\`

File: spec/Knp/DictionaryBundle/Dictionary/CallableDictionarySpec.php [hunk 3]
```diff
@@ -119,8 +128,9 @@ public function execution()
     {
         if (false === $this->executed) {
             $this->executed = true;
+            $args = func_get_args();
```
Comment (Awkan) [inline spec/Knp/DictionaryBundle/Dictionary/CallableDictionarySpec.php]: `\`

File: src/Knp/DictionaryBundle/Dictionary/CallableDictionary.php [hunk 4]
```diff
@@ -21,14 +21,19 @@ class CallableDictionary implements Dictionary
...
+     * @param array    $callableArgs
      */
-    public function __construct($name, callable $callable)
+    public function __construct($name, callable $callable, array $callableArgs = [])
     {
-        $this->name     = $name;
-        $this->callable = $callable;
+        $this->name         = $name;
+        $this->callable     = $callable;
+        $this->callableArgs = $callableArgs;
```
Comment (Awkan) [inline src/Knp/DictionaryBundle/Dictionary/CallableDictionary.php]: indent

Comment (Nek-) [reply src/Knp/DictionaryBundle/Dictionary/CallableDictionary.php]: To fix this, I assume we firstly need to change this: https://github.com/biig-io/DictionaryBundle/blob/master/.php_cs

File: src/Knp/DictionaryBundle/Dictionary/CallableDictionary.php [hunk 5]
```diff
@@ -120,7 +125,7 @@ protected function hydrate()
             return;
         }
 
-        $values = call_user_func($this->callable);
+        $values = call_user_func_array($this->callable, $this->callableArgs);
 
         if (false === is_array($values) && false === $values instanceof \ArrayAccess) {
```
Comment (athos7933) [inline src/Knp/DictionaryBundle/Dictionary/CallableDictionary.php]: \is_array

**Pull request merged**